test/new-e2e/tests/agent-platform/install-script/install_script_test.go

```diff
@@ -186,3 +215,78 @@ func (is *installScriptSuite) DogstatsdAgentTest() {
 	common.CheckInstallationInstallScript(is.T(), client)
 	is.testUninstall(client, "datadog-dogstatsd")
 }
+
+type installScriptSuiteSysVInit struct {
+	e2e.BaseSuite[environments.Host]
+}
+
+func (is *installScriptSuiteSysVInit) TestInstallAgent() {
+	flavor := "datadog-agent"
+	containerName := "installation-target"
+	remoteHost := is.Env().RemoteHost
+	host := NewHostFromDocker(remoteHost, containerName)
+
+	// Start docker container
+	dockerImage := "public.ecr.aws/ubuntu/ubuntu:22.04_stable"
+	_, err := remoteHost.Execute(
+		fmt.Sprintf("docker run -d -e DD_HOSTNAME=docker-test --name '%s' '%s' tail -f /dev/null", containerName, dockerImage),
+	)
+	require.NoError(is.T(), err)
+	defer remoteHost.Execute(fmt.Sprintf("docker rm -f '%s'", containerName))
+
+	fileManager := filemanager.NewUnix(host)
+	agentClient, err := client.NewHostAgentClient(is, remoteHost.HostOutput, false)
+	require.NoError(is.T(), err)
+
+	unixHelper := helpers.NewUnix()
+	client := common.NewTestClient(host, agentClient, fileManager, unixHelper)
+
+	// We need `curl` to download the script, and `sudo` because all the existing test helpers run commands
+	// through it.
+	_, err = client.ExecuteWithRetry("apt-get update && apt-get install -y curl sudo")
+	require.NoError(is.T(), err)
+
+	install.Unix(is.T(), client, installparams.WithArch(*architecture), installparams.WithFlavor(flavor), installparams.WithMajorVersion(*majorVersion))
+
+	common.CheckInstallation(is.T(), client)
+	common.CheckSigningKeys(is.T(), client)
+
+	// We can't run the tests that rely on the `AgentClient` since it's not possible to pass a custom executor
+	// to it (which is necessary in this case, as this is not quite the host case nor quite the docker case).
+	// We run a few selected sanity checks here instead (sufficient for this platform anyway)
+	is.T().Run("datadog-agent service running", func(tt *testing.T) {
+		_, err := client.SvcManager.Status(client.Helper.GetServiceName())
+		require.NoError(tt, err, "datadog-agent service should be running")
+	})
+	is.T().Run("status command no errors", func(tt *testing.T) {
+		statusOutput, err := client.ExecuteWithRetry("datadog-agent \"status\"")
+		require.NoError(is.T(), err)
+
+		// API Key is invalid we should not check for the following error
+		statusOutput = strings.ReplaceAll(statusOutput, "[ERROR] API Key is invalid", "API Key is invalid")
+		require.NotContains(tt, statusOutput, "ERROR")
```

Comment:
Is not apparently clear we we need to replace and assert. Could you explain why did you do it?

Reply:
It's copied from https://github.com/DataDog/datadog-agent/blob/0620b25bd8d9f024675b66ac0c460b6af3013566/test/new-e2e/tests/agent-platform/common/agent_behaviour.go#L70-L76

The reason it's copied is that the existing test helper is hard to reuse (without significant changes elsewhere).

The way I read this, the reason for the replace is to remove the `[ERROR]` from lines that refer to the API Key being invalid (considering those acceptable given the test's context); but we do want to assert that otherwise there are no errors on the status output.